### PR TITLE
Remove prometheus reference from storage

### PR DESCRIPTION
**Pull request by RainbowMango**
**What type of PR is this?**
/kind feature
/priority important-soon

**What this PR does / why we need it**:
Remove direct reference to Prometheus.

To get [metrics stability](https://github.com/kubernetes/enhancements/blob/master/keps/sig-instrumentation/20190404-kubernetes-control-plane-metrics-stability.md) into Beta, we will make it illegal to directly import prometheus methods in Kubernetes components.

**Which issue(s) this PR fixes**:
Refer https://github.com/kubernetes/enhancements/issues/1238

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

### CODE REVIEW

**Comment by jktomer** [comment]
LGTM (but mine doesn't count yet, you'll need an /lgtm from someone else)

**Comment by RainbowMango** [comment]
ping @lavalamp and 
/assign @wojtek-t

https://github.com/kubernetes/kubernetes/pull/83837 rely on this.

**Comment by RainbowMango** [comment]
/test pull-kubernetes-dependencies
Can't access the logs.

**Comment by RainbowMango** [comment]
@wojtek-t 
I made a mistake in last fore push(forgot add vendor-related changes).

This should be fine.

File: staging/src/k8s.io/apiserver/go.mod [hunk 1]
```diff
@@ -29,7 +29,6 @@ require (
 	github.com/munnerz/goautoneg v0.0.0-20191010083416-a7dc8b61c822
 	github.com/pborman/uuid v1.2.0
 	github.com/pquerna/cachecontrol v0.0.0-20171018203845-0dec1b30a021 // indirect
-	github.com/prometheus/client_golang v0.9.4
```
Comment (RainbowMango) [inline staging/src/k8s.io/apiserver/go.mod]: FYI:
Removed this line making robot happy.

**Pull request merged**